Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/DictationHandler.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Script used to start and stop recording sessions in the current dictation system and report the transcribed text via UnityEvents.
+    /// For this script to work, a dictation system like 'Windows Dictation Input Provider' must be added to the Data Providers in the Input System profile.
+    /// </summary>
+    public class DictationHandler : BaseInputHandler, IMixedRealityDictationHandler
+    {
+        [SerializeField]
+        [Tooltip("Time length in seconds before the dictation session ends due to lack of audio input in case there was no audio heard in the current session")]
+        private float initialSilenceTimeout = 5f;
+
+        [SerializeField]
+        [Tooltip("Time length in seconds before the dictation session ends due to lack of audio input.")]
+        private float autoSilenceTimeout = 20f;
+
+        [SerializeField]
+        [Tooltip("Length in seconds for the dictation service to listen")]
+        private int recordingTime = 10;
+
+        [SerializeField]
+        [Tooltip("Whether recording should start automatically on start")]
+        private bool startRecordingOnStart = false;
+
+        [System.Serializable]
+        public class StringUnityEvent : UnityEvent<string> { }
+
+        /// <summary>
+        /// Event raised while the user is talking. As the recognizer listens, it provides text of what it's heard so far.
+        /// </summary>
+        public StringUnityEvent OnDictationHypothesis;
+
+        /// <summary>
+        /// Event raised after the user pauses, typically at the end of a sentence. Contains the full recognized string so far.
+        /// </summary>
+        public StringUnityEvent OnDictationResult;
+
+        /// <summary>
+        /// Event raised when the recognizer stops. Contains the final recognized string.
+        /// </summary>
+        public StringUnityEvent OnDictationComplete;
+
+        /// <summary>
+        /// Event raised when an error occurs. Contains the string representation of the error reason.
+        /// </summary>
+        public StringUnityEvent OnDictationError;
+
+        private bool isRecording = false;
```

Comment:
> private bool isRecording = false; [](start = 7, length = 34)

Is this intended to be used for logging/debugging purposes only? It looks like this is getting set correctly but not inspected.

Reply:
Or was this kinda to make sure that StartRecording doesn't invoke dictationSystem.StartRecording in the case it gets double-called?

---
In reply to: [271491908](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3764#discussion_r271491908) [](ancestors = 271491908)

Reply:
The second. It must have been lost when refactoring. Thanks for the catch!